egui_web/src/lib.rs

```diff
@@ -838,3 +952,34 @@ fn install_canvas_events(runner_ref: &AppRunnerRef) -> Result<(), JsValue> {
 
     Ok(())
 }
+
+fn manipulate_agent(canvas_id: &str, latest_cursor: Option<egui::Pos2>) -> Option<()> {
+    use wasm_bindgen::JsCast;
+    use web_sys::HtmlInputElement;
+    let window = web_sys::window()?;
+    let document = window.document()?;
+    let input: HtmlInputElement = document.get_element_by_id(AGENT_ID)?.dyn_into().unwrap();
+    let cutsor_txt = document.body()?.style().get_property_value("cursor").ok()?;
+    let style = canvas_element(canvas_id)?.style();
+    if cutsor_txt == cursor_web_name(egui::CursorIcon::Text) {
+        input.set_hidden(false);
+        input.focus().ok()?;
+        // Panning canvas so that text edit is shown at 30%
+        // Only on touch screens, when keyboard popups
+        if let Some(p) = latest_cursor {
+            let inner_height = window.inner_height().ok()?.as_f64()?;
+            let new_pos = ((0.8 - p.y / inner_height as f32) * 100.0).floor();
+            if new_pos < 50.0 {
+                style.set_property("position", "absolute").ok()?;
+                let new_pos = new_pos.to_string() + "%";
+                style.set_property("top", &new_pos).ok()?;
+            }
+        }
+    } else {
+        input.blur().ok()?;
+        input.set_hidden(true);
+        style.set_property("position", "absolute").unwrap();
+        style.set_property("top", "50%").ok()?;
```

Comment:
When clicking anywhere (except in a text field), this will move the canvas down 50%. I think this was meant to be:
```suggestion
        style.set_property("position", "absolute").unwrap();
        style.set_property("top", "0%").ok()?;
```

Reply:
EDIT: here is the full fix: https://github.com/emilk/egui/commit/41a3bb6b6287f8217792d0800722cfa2cfb4c13c

Reply:
This is because of #151. I was testing using the old index.html, in which canvas was originally translated 50% above.
Is is ok to commit another PR to create canvas inside wasm insetad of html? Such errors can be reduced.

Reply:
I have a plan to one day allow multiple egui instances in the same HTML page, so that egui can be embedded in normal HTML. That's why I pass in the id of the canvas: it is so we can have multiple canvases (in the future).